.github/workflows/deploy-to-heroku.yml

```diff
@@ -1,15 +1,16 @@
 name: Deploy to Heroku
 
 on:
-  push:
-    branches:
-      - master
+  pull_request:
+    types:
+      - closed
   # Deploys to Heroku once each day
   schedule:
     - cron: "0 0 * * *"
 
 jobs:
-  build:
+  if_merged:
```

Comment:
@vipulgupta2048 , you may as well remove the cron schedule above if we're going to have this condition.  This makes sure that only changes from merges are acted on.